File: VB/DocumentServer_PrintToPDF/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Diagnostics
Imports System.Drawing
Imports System.IO
Imports System.Windows.Forms
Imports DevExpress.XtraRichEdit

Namespace DocumentServerExample

    Public Partial Class Form1
        Inherits Form

        Private fileList As List(Of String) = New List(Of String)()

        Private server As RichEditDocumentServer

        Private counter As PerformanceCounter

        Public Sub New()
            InitializeComponent()
            server = New RichEditDocumentServer()
            textBox1.Text = Application.StartupPath
            AddHandler timer1.Tick, New EventHandler(AddressOf timer1_Tick)
            Dim procName As String = Process.GetCurrentProcess().ProcessName
            counter = New PerformanceCounter("Process", "Working Set - Private", procName)
            ShowMemoryUsage()
        End Sub

        Private Sub PrintToPDF(ByVal filePath As String)
            Dim displayText As String
            Try
                server.LoadDocument(filePath)
            Catch ex As Exception
                server.CreateNewDocument()
                displayText = String.Format("{0:T} Error:{1} -> {2}", Date.Now, ex.Message, filePath) & Environment.NewLine
                textBox2.Text += displayText
                Return
            End Try

            Dim outFileName As String = Path.ChangeExtension(filePath, "pdf")
            Dim fsOut As FileStream = File.Open(outFileName, FileMode.Create)
            server.ExportToPdf(fsOut)
            fsOut.Close()
            displayText = String.Format("{0:T} Done-> {1}", Date.Now, outFileName) & Environment.NewLine
            textBox2.Text += displayText
        End Sub

        Private Sub UpdateFileList(ByVal path As String)
            If Directory.Exists(path) Then
                Dim files As String() = Directory.GetFiles(path, "*.doc?", SearchOption.AllDirectories)
                For Each file As String In files
                    If Not fileList.Contains(file) Then
                        fileList.Add(file)
                        PrintToPDF(file)
                    End If
                Next
            End If
        End Sub

        Private Sub btnConvert_Click(ByVal sender As Object, ByVal e As EventArgs)
            If timer1.Enabled Then
                timer1.Stop()
                btnConvert.Text = "Start!"
                textBox1.ReadOnly = False
            Else
                timer1.Start()
                btnConvert.Text = "Stop!"
                textBox1.ReadOnly = True
            End If
        End Sub

        Private Sub timer1_Tick(ByVal sender As Object, ByVal e As EventArgs)
            UpdateFileList(textBox1.Text)
            ShowMemoryUsage()
        End Sub

        Private Sub ShowMemoryUsage()
            label1.Text = String.Format("Memory usage: {0:N0} K", counter.RawValue / 1024)
        End Sub
    End Class
End Namespace

File: CS/DocumentServer_PrintToPDF/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.IO;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraRichEdit;

namespace DocumentServerExample
{
    public partial class Form1 : Form
    {
        List<string> fileList = new List<string>();
        RichEditDocumentServer server;
        PerformanceCounter counter;

        public Form1()
        {
            InitializeComponent();
            server = new RichEditDocumentServer();
           
            textBox1.Text = Application.StartupPath;
            timer1.Tick += new EventHandler(timer1_Tick);

            string procName = Process.GetCurrentProcess().ProcessName;
            counter = new PerformanceCounter("Process", "Working Set - Private", procName);
            ShowMemoryUsage();
        }

        private void PrintToPDF(string filePath)
        {
            string displayText;

            try {
                server.LoadDocument(filePath);
            }
            catch (Exception ex) {
                server.CreateNewDocument();
                displayText = String.Format("{0:T} Error:{1} -> {2}", DateTime.Now, ex.Message, filePath) + Environment.NewLine;
                textBox2.Text += displayText;
                return;
            }
            string outFileName = Path.ChangeExtension(filePath, "pdf");
            FileStream fsOut = File.Open(outFileName, FileMode.Create);
            server.ExportToPdf(fsOut);
            fsOut.Close();
            displayText = String.Format("{0:T} Done-> {1}",DateTime.Now, outFileName) + Environment.NewLine;
            textBox2.Text += displayText;
        }

        private void UpdateFileList(string path)
        {
            if (Directory.Exists(path)) {
                string[] files = System.IO.Directory.GetFiles(path, "*.doc?", System.IO.SearchOption.AllDirectories);

                foreach (string file in files) {
                    if (!fileList.Contains(file)) {
                        fileList.Add(file);
                        PrintToPDF(file);
                    }
                }
            }
        }

        private void btnConvert_Click(object sender, EventArgs e)
        {
            if (timer1.Enabled) {
                timer1.Stop();
                btnConvert.Text = "Start!";
                textBox1.ReadOnly = false;
            }
            else {
                timer1.Start();
                btnConvert.Text = "Stop!";
                textBox1.ReadOnly = true;
            }
        }

        void timer1_Tick(object sender, EventArgs e)
        {
            UpdateFileList(textBox1.Text);
            ShowMemoryUsage();
        }
        private void ShowMemoryUsage()
        {
            label1.Text = String.Format("Memory usage: {0:N0} K", counter.RawValue/1024);
        }
    }
}